add pact readme WIP

LindseySaari opened the pull request:
Adds a WIP pact readme including implementation details and background info on PACT as a CDCT tool

LindseySaari committed:
add pact readme WIP

LindseySaari committed:
Add header for configuration details section

LindseySaari committed:
Add placeholders for Pact setup and todo items

LindseySaari committed:
update readme with be/fe instructions

LindseySaari committed:
add docker workflow doc updates

LindseySaari committed:
add native setup instructions

LindseySaari committed:
add auth session stubbing info and links

U-DON committed:
Updated FE content in Pact doc.

U-DON commented on teams/vsp/teams/tools/backend/pact/readme.md:
```diff
@@ -1 +1,420 @@
+
+### Configure the pact_uri/broker_url
+
+To work with one pact in the broker vs all pacts, point the `pact_uri` in the `pact_helper.rb` file to a specific contract in the broker (see [here](https://github.com/department-of-veterans-affairs/vets-api/pull/4612/files#diff-79dfb3765c26716457a005b9d343a160R38)), otherwise the rake task will run all the pacts pushed to the [heroku broker](https://vagov-pact-broker.herokuapp.com/).
```
Should this section mention using [`rake pact:verify:at`](https://github.com/pact-foundation/pact-ruby/wiki/Verifying-pacts#verifying-a-pact-at-any-url-using-rake-pactverifyat)?

LindseySaari committed:
remove pool

LindseySaari committed:
update requirements

meganhkelley committed:
change PACT to pact

meganhkelley committed:
move to /platform folder and rename

meganhkelley committed:
add table of contents

LindseySaari committed:
update to use rake pact:verify:at

meganhkelley commented:
Hey y’all @U-DON @LindseySaari thanks so much for putting these docs together! I’ve made a few edits (style, formatting), and have some higher level comments & requests below. 

(1) I added a table of contents — wow! This is a very hefty doc. We don’t need to leave this full table of contents in the final doc (although we could), but it’s a good tool for us to evaluate the formatting. Does the current formatting of this doc make sense when y’all look at that ToC? Should this be broken out into multiple, smaller, cross-linked docs? Should the “how to set up” sections be way higher up and more the focus of the doc? Are things using the right header level? (Header level 5 ends up with a font size smaller than the body text which is not ideal — do we need to nest things that deep?) _I don’t feel like I have enough technical understanding to answer these Qs myself, although would be happy to have a work session with someone where we talk through this stuff, if that would be helpful._

(2) By `Requirements` do we mean this section describes integration test coverage required by VSP? If so let’s make that clear. If there’s a doc we can link to for `required code review process` here let’s do so.
* Check box when addressed [ ]

(3) In the `Broker` section, the `example setup instructions` link to HackMD — where are these supposed to link? Actually, a few other links throughout also link to HackMD (Pact example > Local json Files > details _and_ How to configure the provider codebase > quick reference checklist). Could someone please review the doc for lingering HackMD links? Thanks!
* Check box when addressed [ ]

(4) In `Implementation details` the link to `workflow details` goes here which is a really odd place to send people (the PR for a design doc). What are we trying to convey here? If linking to the design doc is the right call, could we please merge that design doc and link to the actual file? 
* Check box when addressed [ ]

(5) The `Pact example` setup instructions link goes to a really odd place as well — this seems to be a doc with some discovery info for Pact that’s on a branch in a folder that doesn’t make sense. If this is useful info that we want to reference, let’s move it to the `platform/testing/integration-testing` folder — either in a new doc, or we can move this whole doc over if we give it a better name. Also, we should always link to master wherever possible.
* Check box when addressed [ ] 

(6) Are the `Broker` and `Live Pacts` sections duplicates? Or are these conveying different info? 
* Check box when addressed (as applicable) [ ] 

(7) We refer to “the pact helper” a couple times, but never actually explain what this is, how to use it, etc. Is that implied, or does that info need to be called out? 
* Check box when addressed (as applicable) [ ]

U-DON commented:
I think it makes sense to split the FE and BE setup instructions into their own docs and cross link from here. It's likely there will be more content to add in both sections, and I agree the whole thing is already rather hefty as it is now.

There are probably a few things we could probably leave out, like the Pact Broker local setup. VFS teams using Pact should be able to adequately test their pacts by using the locally generated pacts and verifying locally. Only VSP will have to be concerned with maintaining the broker.

LindseySaari commented:
Thanks for the feedback @meganhkelley! Going to work through your feedback points above and push up changes. Thanks for pointing out that the links point to hackmd! Forgot to update those when doing a copy/paste from the original workspace.

LindseySaari committed:
update/remove local broker details

LindseySaari committed:
add pact_helper details

LindseySaari commented:
@U-DON Any thoughts on this point? (4) In Implementation details the link to workflow details goes here which is a really odd place to send people (the PR for a design doc). What are we trying to convey here? If linking to the design doc is the right call, could we please merge that design doc and link to the actual file?

LindseySaari committed:
update requirements

LindseySaari committed:
remove local broker pact example from TOC

LindseySaari committed:
add link for code review process

U-DON commented:
Maybe we could do without that section too,  or at least for now. From a VFS perspective, I think the reason people visit this doc would mainly be to figure out how to use Pact, and the design doc might be too dense to be useful to them.

LindseySaari committed:
remove design doc details

LindseySaari committed:
condense duplicate broker info

LindseySaari committed:
update to plural

LindseySaari committed:
remove live pacts from TOC

Condensed this info into the broker section

LindseySaari committed:
remove dup provider states link & info

LindseySaari committed:
remove hackmd links

LindseySaari committed:
Remove publish flag from docs

Results will only be published to the broker when run in CI (this is now driven from an ENV variable -- set in Circle)

LindseySaari committed:
refactor steps based on Circle integration

gmaciolek committed:
Formatting and organization updates (#13106)

My edits mostly fell into the following areas:
- using parallel structure in headings and bullets
- using active voice
- trimming words to make sentences shorter
- using sentence case
- applying style guide recommendations for formatting

I removed the quick reference checklist in section about setting up the backend. If you want to add it back, I would format it as a checklist and not include the "Broker matrix and tagging" information since that's not really a step.

Co-authored-by: Lindsey Hattamer <lindsey.hattamer@oddball.io>

LindseySaari committed:
add circle info

U-DON committed:
Replaced old Pact test file suffix with new.
